src/Keboola/DbExtractor/Extractor/Extractor.php

```diff
@@ -167,7 +174,11 @@ public function export(array $table): array
         ];
         // output state
         if (!empty($result['lastFetchedRow'])) {
-            $output['state']['lastFetchedRow'] = $result['lastFetchedRow'];
+            if ($maxValue) {
+                $output['state']['lastFetchedRow'] = $maxValue;
+            } else {
+                $output['state']['lastFetchedRow'] = $result['lastFetchedRow'];
+            }
```

Comment:
Sorry, this is still incorrect.  If there is no `OrderBy` then the result will not have a `lastFetchedRow` defined (it will use the `$maxValue` instead).

So we need to instead check if $maxValue is defined, and if so we use that.
See: https://github.com/keboola/db-extractor-mssql/blob/master/src/Keboola/DbExtractor/Extractor/MSSQL.php#L241

Reply:
you mean this?

Reply:
@ondrajodas Not quite!  You are checking `if (!empty[$result]['lastFetchedRow']` but in the case of no limit, we don't use `orderBy` for performance reasons, and `lastFetchedRow` is not available.  It should check here instead whether incrementalFetchingColumn is set.

Reply:
i checking it here
https://github.com/keboola/db-extractor-common/pull/136/commits/6fdebbedba45db70d43cd69da7e55c9f2837a6fc

Reply:
shit, sorry, I saw cache copy, tohle vypada dobre👍